strawberry/quart/views.py

```diff
@@ -46,6 +59,34 @@ async def get_form_data(self) -> FormData:
         return FormData(files=files, form=form)
 
 
+class QuartWebSocketAdapter(AsyncWebSocketAdapter):
+    def __init__(self, view: AsyncBaseHTTPView, request, ws) -> None:
+        super().__init__(view)
+        self.ws = websocket
+
+    async def iter_json(
+        self, *, ignore_parsing_errors: bool = False
+    ) -> AsyncGenerator[object, None]:
+        while True:
+            message = await self.ws.receive()
+            if type(message) is bytes:
+                raise NonTextMessageReceived
+            try:
+                yield self.view.decode_json(message)
+            except JSONDecodeError as e:
+                if not ignore_parsing_errors:
+                    raise NonJsonMessageReceived from e
+
+    async def send_json(self, message: Mapping[str, object]) -> None:
+        try:
+            await self.ws.send(self.view.encode_json(message))
+        except Exception as exc:
+            raise WebSocketDisconnected from exc
+
+    async def close(self, code: int, reason: str) -> None:
+        await self.ws.close(code, reason=reason)
+
+
 class GraphQLView(
     AsyncBaseHTTPView[
         Request, Response, Response, Request, Response, Context, RootValue
```

Comment:
Ideally, we would adjust these types to include `quart.Websocket` and make `create_websocket_response` return `websocket` etc. so that we end up with sound types.

That being said, I recognize `quart` loves their global context vars, so we could go with `None` here too. In that case `get_context` might need an update, since it's currently just using the return value from `create_websocket_response` which is `None`.

Reply:
Yeah, I wasn't quite sure how to approach this exactly because of the context vars, But once the tests are all passing, we can take a look at cleaning this up.